spring-boot/proxy/src/test/java/LocalApiKeyAndServiceApiManagerTest.java

```diff
@@ -0,0 +1,116 @@
+/*-
+ * #%L
+ * coatrack-proxy
+ * %%
+ * Copyright (C) 2013 - 2021 Corizon | Institut für angewandte Systemtechnik Bremen GmbH (ATB)
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+import eu.coatrack.api.ApiKey;
+
+import eu.coatrack.proxy.security.AdminCommunicator;
+import eu.coatrack.proxy.security.LocalApiKeyAndServiceApiManager;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+public class LocalApiKeyAndServiceApiManagerTest extends LocalApiKeyAndServiceApiManager{
```

Comment:
This sub-classing approach looks a little bit like we are testing our manager "from the inside" instead of from an "external black-box view". Also, in the worst case it could happen that this sub-class would work fine, while instances of the super class would not work fine. 

Therefore it would be better to instantiate the "object under test" and test it as a "black box".

Reply:
Good point. Instead of my approach, a fake-AdminCommunicator could be injected via constructor to provide test conditions.

Reply:
I solved this with mocks.